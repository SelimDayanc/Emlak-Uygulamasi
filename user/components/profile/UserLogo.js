/**
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import colors from '../../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export default class UserLogo extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  render() {
    const {user} = this.props;

    return (
      <View style={styles.container}>
        {user.image
          ? <Image
              source={{uri: user.image}}
              style={styles.logo}
              resizeMode="cover"
            />
          : <FontAwesome
              name="picture-o"
              color="white"
              size={200}
              style={styles.emptyImageIcon}
            />}
        <Text style={styles.username}>{user.name}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 10,
  },
  username: {
    fontSize: 20,
    fontWeight: '700',
    color: colors.darkGrey,
    paddingTop: 5,
  },
  logo: {
    height: 100,
    width: 100,
    borderRadius: 50,
  },
});
