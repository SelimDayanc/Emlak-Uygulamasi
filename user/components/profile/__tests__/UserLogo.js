import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserLogo from './../UserLogo';

test('renders heart', () => {
  const tree = renderer
    .create(<UserLogo user={{name: 'name', email: 'z4ls@live.com'}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
