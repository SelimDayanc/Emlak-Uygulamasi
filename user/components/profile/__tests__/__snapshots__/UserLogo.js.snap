// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 200,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï€¾
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#484848",
        "fontSize": 20,
        "fontWeight": "700",
        "paddingTop": 5,
      }
    }
  >
    name
  </Text>
</View>
`;
